datumaro/components/filter.py

```diff
@@ -53,7 +53,8 @@ def encode_image(cls, image):
         ET.SubElement(image_elem, "height").text = str(h)
 
         ET.SubElement(image_elem, "has_data").text = "%d" % int(image.has_data)
-        ET.SubElement(image_elem, "path").text = image.path
+        if getattr(image, "path", None) is not None:
```

Comment:
How about using `hasattr`?
```suggestion
        if hasattr(image, "path"):
```

Reply:
Fixed in https://github.com/openvinotoolkit/datumaro/pull/921/commits/18d8a407ae2cab1e93d02da1bb58b89722d273fe